test/routing_test.rb

```diff
@@ -660,6 +660,20 @@ class RoutingTest < Minitest::Test
     assert ok?
   end
 
+  it 'unescapes named parameters and splats' do
+    mock_app {
+      get '/:foo/*' do |a, b|
+        assert_equal "foo\x80", params['foo']
+        assert_predicate params['foo'], :valid_encoding?
```

Comment:
This fails because params are force encoded with the default encoding here:

https://github.com/sinatra/sinatra/blob/master/lib/sinatra/base.rb#L1026

Reply:
Just droping the permalink version of that link here as the line it points to might change in the future: https://github.com/sinatra/sinatra/blob/1b35eaa7b8296313d48ec27e52955f10f35462fb/lib/sinatra/base.rb#L1026